core/src/main/scala/kafka/admin/TopicCommand.scala

```diff
@@ -212,9 +213,10 @@ object TopicCommand extends Logging {
             for ((partitionId, assignedReplicas) <- sortedPartitions) {
               val inSyncReplicas = zkUtils.getInSyncReplicasForPartition(topic, partitionId)
               val leader = zkUtils.getLeaderForPartition(topic, partitionId)
-              if ((!reportUnderReplicatedPartitions && !reportUnavailablePartitions) ||
+              if ((!reportUnderReplicatedPartitions && !reportUnavailablePartitions && !reportUnbalancedPartitions) ||
```

Comment:
I think this is the intention of all these long checks...

```
if just describe topic:
  for all topics:
    for all partitions:
      describeConfig
      describePartition

if describe topic with describeConfig and no (under-replicated, unavailable):
  for all topics:
    for all partitions:
      describeConfig

if describe topic for only under-replicated partitions:
  for all topics:
    for all partitions:
      if under-replicated partition:
        describePartition

if describe topic for only unavailable partitions:
  for all topics:
    for all partitions:
      if unavailable partition:
        describePartition

if describe topic for under-replicated and unavailable partitions:
  for all topics:
    for all partitions:
      if unavailable or under-replicated partition:
        describePartition
```

However, it is achieved in a complicated way.